src/Microsoft.IdentityModel.Tokens/JsonWebKeySet.cs

```diff
@@ -117,73 +139,150 @@ public JsonWebKeySet(string json, JsonSerializerSettings jsonSerializerSettings)
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore, NullValueHandling = NullValueHandling.Ignore, PropertyName = JsonWebKeySetParameterNames.Keys, Required = Required.Default)]
         public IList<JsonWebKey> Keys { get; private set; } = new List<JsonWebKey>();
 
+        /// <summary>
+        /// Flag that controls whether unresolved JsonWebKeys will be included in the resulting collection of <see cref="GetSigningKeys"/> method.
+        /// </summary>
+        [DefaultValue(true)]
+        public static bool SkipUnresolvedJsonWebKeys { get; set; } = true;
+
         /// <summary>
         /// Returns the JsonWebKeys as a <see cref="IList{SecurityKey}"/>.
         /// </summary>
+        /// <remarks>
+        /// To include unresolved JsonWebKeys in the resulting <see cref="SecurityKey"/> collection, set <see cref="SkipUnresolvedJsonWebKeys"/> to <c>false</c>.
+        /// </remarks>
         public IList<SecurityKey> GetSigningKeys()
         {
-            List<SecurityKey> keys = new List<SecurityKey>();
-            for (int i = 0; i < Keys.Count; i++)
-            {
-                JsonWebKey webKey = Keys[i];
+            var signingKeys = new List<SecurityKey>();
 
-                if (!StringComparer.Ordinal.Equals(webKey.Kty, JsonWebAlgorithmsKeyTypes.RSA))
+            foreach (var webKey in Keys)
+            {
+                // skip if "use" (Public Key Use) parameter is not empty or "sig"
+                // https://tools.ietf.org/html/rfc7517#section-4.2
+                if (!(string.IsNullOrWhiteSpace(webKey.Use) || webKey.Use.Equals(JsonWebKeyUseNames.Sig, StringComparison.Ordinal)))
```

Comment:
This could be breaking for some users. Should we apply the SkipUnresolvedJsonWebKeys flag?

Reply:
We had this check before the change.
I guess that we could include these keys as well when the flag is set to false.

Reply:
addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1143/commits/3f39974831857fc0e2abe7afc477ba17caa1c359.

Reply:
addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1143/commits/3f39974831857fc0e2abe7afc477ba17caa1c359